cli/src/commands/init.js

```diff
@@ -183,7 +186,42 @@ const handler = async ({ force, name, cwd, lib }) => {
         })
     }
 
-    const entrypoint = lib ? 'src/index.jsx' : 'src/App.jsx'
+    if (typeScript) {
+        // copy tsconfig
+        reporter.info('Copying tsconfig')
+        fs.copyFileSync(paths.initTSConfig, paths.tsConfig)
+
+        reporter.info('install TypeScript as a dev dependency')
+        // ToDO: restrict the major version of TS we install
+        await exec({
+            cmd: 'yarn',
+            args: ['add', 'typescript', '--dev'],
+            cwd: paths.base,
+        })
+
+        // Todo: check why d2 start is renaming TS files to JS
```

Comment:
Is this still happening? I haven't noticed it

Note that some files from the app shell will be copied in to `.d2/shell/`, which are called `App.jsx` and `App.test.jsx`, which may be confusing:

<img width="335" alt="Screenshot 2024-08-05 at 5 32 38 PM" src="https://github.com/user-attachments/assets/e0142cab-5d2b-47a5-ad56-8d90f0012dd3">